### ISSUE TITLE
Loop in Compose Message flow for users with no providers 

### ISSUE THREAD

**Issue by bridgethapner** [department-of-veterans-affairs/va.gov-team#26541]
## What happened?

- When a user taps "Compose a Message" from their Secure Messaging Inbox and is not registered with providers, they see a screen with an alert saying "We can't match you with a provider" rather than the compose form. 
- If they hit "Cancel" from this screen, they are routed to a second alert asking them if they want to cancel their message or continue editing. 
- If they hit "Back" from this screen they are taken back to the error screen. 
- _Note: the "Go to inbox" on the "We can't match you with a provider"  screen takes users back to the Inbox successfully, but there is potential for someone to get stuck in a loop using the top nav bar items._ 

![sm-no-providers-bug.png](https://images.zenhubusercontent.com/5d8e52f61c53fd000173c5de/ba2fc0c5-396e-4013-84cd-8fbd19b378fd)

## Specs:
- Device: iPhone 12 mini
- Test User: judy.morrison@id.me

## Steps to Reproduce
- Go to the Secure Messaging Inbox on a test user that is registered for Secure Messaging but has no providers (judy.morrison@id.me is one) 
- Tap "Compose a message"
- Tap "Cancel" on the screen that says "We can't match you with a provider" 
- Then you should see the "Cancel this message?" screen
- Hit "Back" on the "Cancel this message?" screen, and you will be taken back to "We can't match you with a provider"

## Desired behavior
If a user gets an error alert upon navigating to the Reply or Compose screen, the "Cancel" button should route back to the previous screen without a confirmation alert. 

## Acceptance Criteria
- [ ] When a user hits "Cancel" from an alert screen return them back to the previous screen without a confirmation.

**Comment by digitalken** [comment]
7/7/2021 - @bridgethapner - I'm logging in with +262 (judy isn't working correctly in messages for me today).  +262 shows that I have no messages but when I tap to compose a message, I'm not presented with the screen about not being matched with a provider.  If I tap Cancel on the compose screen, I'm given the "Cancel this message?" screen.  If I tap back on that screen, I'm sent back to the compose screen.  If I hit cancel on the compose screen, I'm back to the "Cancel this message?" screen ad infinitum with your loop again.  Is this the same bug not yet fixed or is this a new one?  (Also, I'm wondering if this for sure made it into the last build.  Maybe I should wait until the next build comes out to test for sure?) -Ken

https://images.zenhubusercontent.com/133843125/a8a823ea-1887-4bab-a701-ff7d27916269/repeatadinfinitum.mp4

**Comment by drjecker** [comment]
@digitalken I will chime in here. Not sure I am 100% following but I don't think this is a bug. I've tried to capture how the flow should work in this image:

![Screen Shot 2021-07-08 at 3.27.08 PM.png](https://images.zenhubusercontent.com/602e6bbb345265c801d416d8/5ced8428-825c-4c32-a05b-c7b19c2c5d69)

Of note, this flow will be updated as a result of Drafts (which is partially in the build I am looking at) so this will be changing soon. 

So, if the flow you are experiencing reflects what I outlined above then it's fine (if not slightly annoying). If not let me know and I will follow up to make sure I understand.
